app/src/main/java/com/vci/vectorcamapp/incomplete_session/presentation/IncompleteSessionViewModel.kt

```diff
@@ -28,10 +29,24 @@ class IncompleteSessionViewModel @Inject constructor(
     private val _incompleteSessions = sessionRepository.observeIncompleteSessions()
         .stateIn(viewModelScope, SharingStarted.WhileSubscribed(), emptyList())
     private val _state = MutableStateFlow(IncompleteSessionState())
-    val state = combine(_incompleteSessions, _state) { incompleteSessions, state ->
-        state.copy(
-            sessions = incompleteSessions
-        )
+
+    val state = combine(_incompleteSessions, _state) { incompleteSessions, currentState ->
+        val filteredSessions = if (currentState.searchQuery.isBlank()) {
+            incompleteSessions
+        } else {
+            incompleteSessions.filter { session ->
+                val fieldsForSearch = buildList {
+                    add(session.collectorName)
+                    add(session.collectorTitle)
+                    add(session.houseNumber)
+                    add(session.collectionMethod)
+                    add(session.specimenCondition)
```

Comment:
I would recommend against searching by fields which are not visible to the user. CollectionMethod and SpecimenCondition are not visible in the incomplete session list